oauth2_http/java/com/google/auth/oauth2/UserAuthorizer.java

```diff
@@ -291,9 +320,11 @@ public UserCredentials getCredentialsFromCode(
     GenericData tokenData = new GenericData();
     tokenData.put("code", code);
     tokenData.put("client_id", clientId.getClientId());
-    tokenData.put("client_secret", clientId.getClientSecret());
     tokenData.put("redirect_uri", resolvedCallbackUri);
     tokenData.put("grant_type", "authorization_code");
+    if (this.clientAuthenticationType == ClientAuthenticationType.CLIENT_SECRET_POST) {
+      tokenData.put("client_secret", clientId.getClientSecret());
```

Comment:
Can you add a comment that explains the following

* STS will reject requests that contains this in the POST body.
* This is to maintain existing behavior for existing code.